src/oceans/components/common/Guide.jsx

```diff
@@ -1,28 +1,98 @@
+import React from 'react';
+import Radium from 'radium';
+import Typist from 'react-typist';
+
+import {getState, setState} from '@ml/oceans/state';
+import guide from '@ml/oceans/models/guide';
+import soundLibrary from '@ml/oceans/models/soundLibrary';
+import styles from '@ml/oceans/styles';
+import colors from '@ml/oceans/styles/colors';
+import I18n from '@ml/oceans/i18n';
+import {Button} from '@ml/oceans/components/common';
+import arrowDownImage from '@public/images/arrow-down.png';
+import {
+  startTextToSpeech,
+  stopTextToSpeech,
+  hasTextToSpeechVoices
+} from '@ml/utils/TextToSpeech';
+
+export const stopTypingSounds = () => {
+  const state = getState();
+  if (state.guideTypingTimer) {
+    clearInterval(state.guideTypingTimer);
+    setState({guideTypingTimer: undefined}, {skipCallback: true});
+  }
+};
 
 let UnwrappedGuide = class Guide extends React.Component {
-  onShowing() {
+  onTypingDone() {
     clearInterval(getState().guideTypingTimer);
-    setState({guideShowing: true, guideTypingTimer: null});
+    setState({guideShowing: true, guideTypingTimer: undefined});
   }
 
-  dismissGuideClick() {
-    const dismissed = guide.dismissCurrentGuide();
-    if (dismissed) {
-      soundLibrary.playSound('other');
+  onGuideClick = () => {
+    const state = getState();
+    const currentGuide = guide.getCurrentGuide();
+
+    if (this.updateTextToSpeech(true)) {
+      // This click started text to speech.
+      setState(
+        {
+          hasTextToSpeechStartedByClick: true,
+          textToSpeechCurrentGuide: currentGuide
+        },
+        {skipCallback: true}
+      );
+    } else {
+      // This click did not start text to speech, so attempt
+      // to dismiss the guide.
+      const dismissed = guide.dismissCurrentGuide();
+      if (dismissed) {
+        if (state.textToSpeechLocale) {
+          stopTextToSpeech();
+        }
+        soundLibrary.playSound('other');
+      }
     }
-  }
+  };
+
+  // Called from both the guide click handler and the render method, and
+  // attempts to play text to speech if needed.  Returns true if it believes
+  // it started text to speech.
+  updateTextToSpeech = inClickHandler => {
+    const state = getState();
+    const currentGuide = guide.getCurrentGuide();
+
+    // Do nothing if text to speech is not desired or yet available.
+    if (!state.textToSpeechLocale || !hasTextToSpeechVoices()) {
+      return false;
+    }
+
+    // Do nothing if there is no current guide, or if we've already started
+    // text to speech for the current guide (which might have finished
+    // playing by now).
+    if (
+      !currentGuide ||
+      state.textToSpeechCurrentGuide === currentGuide
+    ) {
+      return false;
+    }
+
+    // If we are in a click handler, do nothing if we've already started
+    // text to speech from a click handler.
+    // If we are not in a click handler, do nothing if we've never started
```

Comment:
Took me a minute to follow here -- I think this bit I was missing in understanding the second case is that we always want to initiate TTS for the first time via click (right)? Could we update this comment to note that?

Reply:
Good call.  Updated.